### PR TITLE
[Develop] updated documents

### PR DESCRIPTION
**Pull request by yunchu**
Updated documents
README.md
- added 'overview' section to introduce OTE core features (inclusive of features for the future release)
- added 'roadmap' section to provide rough estimation of the new features
- added 'issues/discussions' section to guide how to share things for the user
QUICK_START_GUIDE.md
- clarify supported environment
- reviewed all instructions then updated with actual current status

Modified some CLI modules
- updated some help messages for cli commands
- fixed some bugs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,45 +5,92 @@
...
 Please read the [Contribution guide](CONTRIBUTING.md) before starting work on a pull request.
 
+---
+
 ## Known limitations
 
-Training, export, and evaluation scripts for TensorFlow- and most PyTorch-based models from the [misc](#misc) branch are, currently, not production-ready. They serve exploratory purposes and are not validated.
+Training, export, and evaluation scripts for TensorFlow- and most PyTorch-based models from the [misc](https://github.com/openvinotoolkit/training_extensions/tree/misc) branch are, currently, not production-ready. They serve exploratory purposes and are not validated.
 
 ---
```
Comment (yunchu) [inline README.md:93]: @goodsong81 can you check this updates especially "Intro" and "Roadmap" section?

File: QUICK_START_GUIDE.md [hunk 2]
```diff
@@ -2,135 +2,140 @@
 
 ## Prerequisites
 
-- Ubuntu 18.04 / 20.04
-- Python 3.8+
-- for training on GPU: [CUDA Toolkit 11.1](https://developer.nvidia.com/cuda-11.1.1-download-archive)
+Current version of project was tested under following environments
```
Comment (goodsong81) [inline QUICK_START_GUIDE.md]:
```suggestion
Current version of OTE was tested under following environments
```

File: QUICK_START_GUIDE.md [hunk 3]
```diff
@@ -2,135 +2,140 @@
...
-   Although they are not required, You may also want to use Jupyter notebooks or OTE CLI tools:
+   (Optional) You may also want to use Jupyter notebooks or OTE CLI tools:
 
    ```
-   pip3 install notebook; cd ote_cli/notebooks/; jupyter notebook
+   $ pip3 install notebook; cd ote_cli/notebooks/; jupyter notebook
    ```
 
-3. Search for available scripts that create python virtual environments for different task types:
+1. There available scripts that create python virtual environments for different task types:
```
Comment (goodsong81) [inline QUICK_START_GUIDE.md]: There are? Some words might be missing.

File: QUICK_START_GUIDE.md [hunk 4]
```diff
@@ -2,135 +2,140 @@
...
-   ```
+     - Classification
+     - Detection
+     - Segmantation
 
-   Note that the variable `TASK_ALGO_DIR` is set in this example for simplicity and will not be used in scripts.
+   - `external/anomaly/init_venv.sh` can be used to create a virtual environment for the following task types.
+     - Anomaly-classification
+     - Anomaly-detection
+     - Anomaly-segmentation
```
Comment (goodsong81) [inline QUICK_START_GUIDE.md]: Task type should follow the original TaskType enum in ote_sdk.entity.model_template.
Could you add i-seg / rotated det, and fix anomaly-cls -> anomaly_cls, etc?

File: README.md [hunk 5]
```diff
@@ -5,45 +5,92 @@
 [![mypy](https://img.shields.io/badge/%20type_checker-mypy-%231674b1?style=flat)]()
 [![openvino](https://img.shields.io/badge/openvino-2021.4-purple)]()
 
-OpenVINO™ Training Extensions provide a convenient environment to train
-Deep Learning models and convert them using the [OpenVINO™
-toolkit](https://software.intel.com/en-us/openvino-toolkit) for optimized
-inference.
+> **_DISCLAIMERS_**: Some features described below are under development. You can find more detailed estimation from the [Roadmap](#roadmap) section below.
```
Comment (goodsong81) [inline README.md]:
```suggestion
> **_DISCLAIMERS_**: Some features described below are under development (refer to feature/otx branch). You can find more detailed estimation from the [Roadmap](#roadmap) section below.
```

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Thank you for update! Could you revise a little bit?

File: QUICK_START_GUIDE.md (hunk 4 above)
Comment (yunchu) [reply QUICK_START_GUIDE.md]: updated

File: QUICK_START_GUIDE.md [hunk 6]
```diff
@@ -2,135 +2,144 @@
...
-ote find --root $TASK_ALGO_DIR
-```
-
-Output for the mmdetection used in the above example looks as follows:
+usage: ote find [-h] [--root ROOT] [--task_type TASK_TYPE] [--experimental]
 
+optional arguments:
+  -h, --help              show this help message and exit
+  --root ROOT             A root dir where templates should be searched.
+  --task_type TASK_TYPE   filter with the task type (e.g., classification)
```
Comment (sungmanc) [inline QUICK_START_GUIDE.md]: How about showing the overall candidates of `--task_type`? Maybe users aren't aware of overall candidates.

Although users can find `TASK_TYPE` by using `ote find`, if we give the lists, it is more comfortable.

**Review by goodsong81 — DISMISSED** [review]
LGTM.

File: QUICK_START_GUIDE.md (hunk 6 above)
Comment (yunchu) [reply QUICK_START_GUIDE.md]: thanks for the opinion.

File: README.md [hunk 7]
```diff
@@ -5,45 +5,89 @@
...
+- Integrated efficient hyper-parameter optimization
+  - OTE has an integrated, efficient hyper-parameter optimization module. So, you don't need to worry about searching right hyper-parameters. Through dataset proxy and built-in hyper-parameter optimizer, you can get much faster hyper-parameter optimization compared to other off-the-shelf tools. The hyperparameter optimization is dynamically scheduled based on your resource budget.
+- Support widely-used annotation formats
+  - OTE uses [datumaro](https://github.com/openvinotoolkit/datumaro), which is designed for dataset building and transformation, as a default interface for dataset management. All supported formats by datumaro are also consumable by OTE without the need of explicit data conversion. If you want to build your own custom dataset format, you can do this via datumaro CLI and API.
```
Comment (sungmanc) [inline README.md:24]: Current OTE doesn't support the Datumoar as a Data frontend. How about moving this paragraph to the roadmap section?

Comment (yunchu) [reply README.md:24]: as mentioned in the disclaimers on top of the doc, some features introduced in the overview are under development. so the datumaro integration would be expected as under development feature and user can expect that will be available in v1.0.0 by referring the roadmap section as well.

File: QUICK_START_GUIDE.md (hunk 6 above)
Comment (yunchu) [reply QUICK_START_GUIDE.md]: updated. please check it again.

**Review by goodsong81 — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]
LGTM

**Comment by negvet** [comment]
@yunchu I think this breaks `misc/test_docs.py::TestDocs::test_help_stdoutputs_of_tools` because of `(mpa) ...$` inserted into [QUICK_START_GUIDE.md](https://github.com/openvinotoolkit/training_extensions/pull/1383/files#diff-68cee7441fded5317228046160956565dbcd2d9b341714e6fe5bfafeb98afc3d)

**Comment by yunchu** [comment]
@negvet thanks for the input. will take a look.

**Pull request merged**